plugins/logs/plugin.go

```diff
@@ -975,42 +988,33 @@ func (p *Plugin) bufferChunk(buffer *logBuffer, bs []byte) {
 }
 
 func (p *Plugin) maskEvent(ctx context.Context, txn storage.Transaction, input ast.Value, event *EventV1) error {
```

Comment:
A benefit of the old mutex-based approach was that on error, the `PrepareForEval()` call would be retried on subsequent mask/drop calls. Now, if the first (and only) call to `PrepareForEval()` fails for the current configuration, we'll end up with a broken `PreparedEvalQuery` that'll cause a panic when used.

I've made a commit to your branch with a proposed fix and tests asserting the behavior. The fix simply re-emits the error for subsequent calls. I think it's unlikely that subsequent `PrepareForEval()` calls would have a different outcome, so we don't need to retain the old behavior where we retry.

Thoughts? 
This change shouldn't have an impact on performance, but if you're running separate tests on your end, please let us know how this fares.